pkg/controllers/node/controller.go

```diff
@@ -45,50 +51,84 @@ func NewController(kubeClient client.Client) *Controller {
 type Controller struct {
 	kubeClient client.Client
 	readiness  *Readiness
+	expiration *Expiration
 	finalizer  *Finalizer
 }
 
 // Reconcile executes a reallocation control loop for the resource
 func (c *Controller) Reconcile(ctx context.Context, req reconcile.Request) (reconcile.Result, error) {
 	ctx = logging.WithLogger(ctx, logging.FromContext(ctx).Named("Node"))
-	// 1. Retrieve node from reconcile request
+	// 1. Retrieve Node, ignore if not provisioned or terminating
 	stored := &v1.Node{}
 	if err := c.kubeClient.Get(ctx, req.NamespacedName, stored); err != nil {
 		if errors.IsNotFound(err) {
 			return reconcile.Result{}, nil
 		}
-		return reconcile.Result{}, err
+		return result.RetryIfError(ctx, err)
 	}
 	if _, ok := stored.Labels[v1alpha3.ProvisionerNameLabelKey]; !ok {
 		return reconcile.Result{}, nil
 	}
+	if !stored.DeletionTimestamp.IsZero() {
+		return reconcile.Result{}, nil
+	}
 
-	// 2. Execute node reconcilers
+	// 2. Retrieve Provisioner
+	provisioner := &v1alpha3.Provisioner{}
+	if err := c.kubeClient.Get(ctx, types.NamespacedName{Name: stored.Labels[v1alpha3.ProvisionerNameLabelKey]}, provisioner); err != nil {
+		return reconcile.Result{}, err
+	}
```

Comment:
Do we want to do the same error handling logic here as with the node? If it does exist but failed another way, do we want to return `result.RetryIfError(ctx, err)`?

Reply:
+1